pkgs/applications/science/programming/scyther/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, callPackage, callPackage_i686, buildEnv
+, includeGUI ? true
+, version ? "v1.1.3"
+}:
+let
+  source = (callPackage ./sources.nix {}).getSource "${version}";
+
+  cli = callPackage_i686 ./cli.nix { inherit source; };
+  gui = callPackage ./gui.nix { inherit source cli; };
+in
+  buildEnv {
+    name = "scyther-${version}";
+
+    paths = [ cli ] ++ lib.optional includeGUI gui;
+    pathsToLink = ["/bin"];
+
+    postBuild = if includeGUI then ''
```

Comment:
I did a double-take here. I think I understand what's happening; if the GUI's included, then link the GUI protocols in; otherwise, copy the (bare-bones?) protocols from the source package. But I'm not sure why. Can we get a comment explaining this trickery?